src/skeletonization.jl

```diff
@@ -1,81 +1,130 @@
+	corner_score = zeros(Integer, size(img) .+ 2)
 	score_table = [256 128 64; 32 16 8; 4 2 1]
 
 	for i = 2:size(img)[1] + 1
 		for j = 2:size(img)[2] + 1
 			if img[i - 1, j - 1] 
 				corner_score[i - 1:i + 1, j - 1:j + 1] = corner_score[i - 1:i + 1, j - 1:j + 1] + score_table
-            end
+			end
 		end
 	end
 	corner_table[corner_score[2: size(img)[1] + 1, 2: size(img)[2] + 1] .+ 1]
 end
 
-function inner_skeleton_loop(img::AbstractArray{Bool, 2}, order::Array{Int, 1}, table::AbstractArray{Bool, 1})
-    index = findall(img)[order]
-    score_table = [256 128 64; 32 16 8; 4 2 1]
-    result = zeros(Int, size(img))
-    padded_image = zeros(Int, size(img) .+ 2)
-	padded_image[2: size(img)[1] + 1, 2: size(img)[2] + 1] = img
-    for i in index
-        result[i] = table[sum(padded_image[i[1]:i[1] + 2, i[2]:i[2] + 2] .* score_table) + 1]
-        padded_image[i[1] + 1, i[2] + 1] = result[i]
-    end
-    result
-end 
-
-function skeletonize(img::AbstractArray{Bool}, algo::SkeletonizationAlgo)
-    skeletonize_impl(img, algo)
+# traverse foreground and reduce image to skeleton
+function inner_skeleton_loop(img::AbstractArray{Bool, 2}, order::AbstractArray{T, 1}, table::AbstractArray{Bool, 1}) where T <: Integer
+	index = findall(img)[order]
+	score_table = [256 128 64; 32 16 8; 4 2 1]
+	result = falses(size(img))
+	padded_image = padarray(img, Fill(0, (1, 1), (1, 1)))
+	for i in index
+		result[i] = table[sum(padded_image[i[1] - 1:i[1] + 1, i[2] - 1:i[2] + 1] .* score_table) + 1]
+		padded_image[i[1], i[2]] = result[i]
+	end
+	result
 end
 
-function skeletonize_impl(img::AbstractArray{Bool, 2}, algo::MedialAxis)
+"""
+```
+skeletonize(img, MedialAxisTransform())
+```
+
+Returns Medial axis of binary image that matches the dimensions of the input binary image.
+
+# Details
+Skeletonization is a process for reducing foreground regions in a binary image to a
+skeletal remnant that largely preserves the extent and connectivity of the original
+region while throwing away most of the original foreground pixels.
+
+Steps of the algorithm :
+- A look up table is built to check whether a pixel is to be removed or not. The pixel
+is removed if it has more than one neighbour and if removing it doesn't change the connectedness.
+- Distance transform is computed as well as degree of cornerness.
+- Foreground pixels are traversed as per distance transform and cornerness.
+
+# Arguments
+The img parameter needs to be a two-dimensional array of bool type where true represents
+foreground and false represents background.
+
+# Example
+
+Compute the skeleton of rectangular object.
+```julia
+
+using Images, ImageMorphology
+
+img = Bool.([0 0 0 0 0
+			 0 1 1 1 0
+			 0 1 1 1 0
+			 0 1 1 1 0
+			 0 1 1 1 0
+			 0 0 0 0 0])
+
+skeleton = skeletonize(img, MedialAxisTransform())
+
+skeleton = [0 0 0 0 0
+			0 1 0 1 0
+			0 0 1 0 0
+			0 0 1 0 0
+			0 1 0 1 0
+			0 0 0 0 0]
+```
+
+# References
+[1] http://homepages.inf.ed.ac.uk/rbf/HIPR2/skeleton.htm
```

Comment:
add a title to the reference item for clarity

also here are some thoughts on simplifying documentation of function with multiple methods.

https://github.com/JuliaImages/Images.jl/issues/790#issuecomment-478285863
https://github.com/zygmuntszpak/ImageBinarization.jl/pull/25

It'll be great to hear thought from you.

Reply:
Noted. I will keep that in my mind if any other skeletonization algorithm will be implemented in future?